### PR TITLE
backend-plugin-api: update module ID format

### PR DESCRIPTION
**Pull request by Rugvip**
Suggesting this as the path forwards.

See https://github.com/backstage/backstage/pull/13859#discussion_r985789866

### CODE REVIEW

**Review by jhaals — APPROVED** [review]

**Comment by jhaals** [comment]
api reportz

File: packages/backend-plugin-api/src/wiring/factories.ts [hunk 1]
```diff
@@ -69,7 +69,16 @@ export interface BackendModuleConfig<TOptions> {
...
-/** @public */
+/**
+ * @public
+ *
+ * Creates a new backend module for a given plugin.
+ *
+ * The `moduleId` should be equal to the module-specific prefix of the exported name, such
+ * that the full name is `<moduleId><PluginId>Module`.
+ *
+ * The `pluginId` should exactly match the `id` of the plugin that the module extends.
```
Comment (freben) [inline packages/backend-plugin-api/src/wiring/factories.ts:80]: I think this one warrants an extensive example, that includes the relevant module names and ids and variable names used. It's a bit hard to follow just the text here to make a naming decision, even for someone who's involved :)
